### PR TITLE
Vtcca 2966 hotfix

### PR DESCRIPTION
**Pull request by jon-pike**

### CODE REVIEW

File: app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala [hunk 1]
```diff
@@ -46,6 +49,16 @@ class AuthenticatedActionBuilder @Inject()(
           case (None, Some(_)) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID")))
           case (None, None) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID_OR_GROUP_ID")))
         }
+    }recoverWith {
+      case e: InternalError => {
```
Comment (DanielConnelly) [inline app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: if you are throwing all at the end you dont need this.

File: app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala [hunk 2]
```diff
@@ -46,6 +49,16 @@ class AuthenticatedActionBuilder @Inject()(
...
+        logger.warn(e.getMessage)
+        throw e
+      }
+      case e: AuthorisationException => {
```
Comment (DanielConnelly) [inline app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: is AuthorisationException the parent trait of all the errors we seen ?

File: app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala (hunk 1 above)
Comment (jon-pike) [reply app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: Internal error is a child of AuthorisationException

File: app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala (hunk 2 above)
Comment (jon-pike) [reply app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: yeah - happy with this approach?

Comment (DanielConnelly) [reply app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: Yes,

File: app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala (hunk 1 above)
Comment (DanielConnelly) [reply app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala]: ok, that one is fine then.

**Comment by DanielConnelly** [comment]
LGTM 👍

**Pull request merged**